subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc

```diff
@@ -65,6 +65,13 @@ The legacy `IncrementalTaskInputs` API has been removed. For more information, s
 
 === Potential breaking changes
 
+=== Remove implicit `--add-opens` for Gradle workers
+Prior to Gradle 8.0, JDK modules `java.base/java.util` and `java.base/java.lang` were automatically opened in Gradle workers on JDK9+ by passing `--add-opens` CLI arguments. This meant any code executed in a Gradle worker was able to perform deep reflection on JDK internals without warning or failing. These implicit arguments have been removed from all workers and are no longer added by default.
+
+Gradle uses workers internally for several tasks, including the code-quality plugins (Checkstyle, CodeNarc, Pmd), ScalaDoc, AntlrTask, and the JVM compiler daemons. Further, tasks executed using process isolation via the link:worker_api.html[Worker API] are also affected. If the tool in question, any custom extensions to these tools, or plugins which utilize the worker API directly attempt to perform deep reflection into JDK internals during task execution, new warning or errors may appear.
+
+In most cases, these errors can be resolved by updating the code or dependency performing the illegal access. This may include updating code-quality tool versions, updating annotation processor versions, or updating any Gradle plugins which use the worker API. For some examples on possible error or warning outputs which may arise due to this change, see <<remove_test_add_opens>>.
```

Comment:
```suggestion
Before Gradle 8.0, Gradle workers on JDK9+ automatically opened JDK modules `java.base/java.util` and `java.base/java.lang` by passing `--add-opens` CLI arguments. This enabled code executed in a Gradle worker to perform deep reflection on JDK internals without warning or failing. Workers no longer use these implicit arguments.

This effects all internal Gradle workers, which are used for a variety of tasks:

- code-quality plugins (Checkstyle, CodeNarc, Pmd)
- ScalaDoc
- AntlrTask
- JVM compiler daemons
- tasks executed using process isolation via the link:worker_api.html[Worker API] 

New warnings and errors may appear in any tools, extensions, or plugins that perform deep reflection into JDK internals with the worker API.

These errors can be resolved by updating the violating code or dependency. Updates may include:

- code-quality tools
- annotation processors
- any Gradle plugins which use the worker API

For some examples on possible error or warning outputs which may arise due to this change, see <<remove_test_add_opens>>.
```

Suggestion: reorder some sentences to make cause/effect clearer, use bulleted lists for 2+ item inline lists, trim extra words and sentences.